### DISCUSSION TITLE
Randomization across blocks

### DISCUSSION THREAD

**Discussion by alberteseeberg** [jspsych/jsPsych#2623] [Q&A] [answered]
Hi! 😊  
I am making an online experiment, in which each trial consists of listening to an audio file and completing a rating. I have 90 trials in total. I would like to divide these into 4 blocks, and then present participants with a nonsense trial in between blocks for attention check. I have most of it working fine, but I am struggling with the randomization - I would like to randomize all 90 sound files across the 4 different blocks so that each block won't contain the same audio files every time a participant does the experiment, but I only want each sound file to occur once. Does anyone know how to do this?

tia,
Alberte

**Comment by jodeleeuw** [comment, marked as answer]
Hi @alberteseeberg,

One approach would be to create an array of the 90 sound files and shuffle the order:

```js
const sounds = ['sound1.mp3', 'sound2.mp3', ... ];
const randomized_sounds = jsPsych.randomization.shuffle(sounds);
```

Then you can take the first set of `randomized_sounds` to create the first block.

```js
const block_1 = {
  timeline: []
}

// here I'm assuming 23 trials in the first block
for(let i=0; i<23; i++){
  block_1.timeline.push({
    type: jsPsychAudioKeyboardResponse,
    stimulus: randomized_sounds[i]
  });
}
```

... and so on for the other blocks.

Then you can add `block_1`, `block_2`, etc. to your main timeline.

There are other ways you could do this as well, so feel free to share more of your current implementation if you'd like a suggestion that more closely matches what you currently are working with.

**Reply by alberteseeberg** [reply]
Hi @jodeleeuw! 

Thank you for your reply - I appreciate it!
I also just found a way to do this now without dividing them into blocks actually, using a conditional timeline - like this:

<pre> 
/*create a count trial*/
var trial_count = 0

/*create a conditional timeline for the random questions*/
var nonsense_conditional = {
  timeline: [nonsense],
  conditional_function: function() {
    //increment trial count - in the first run through the timeline variables procedure, trial_count will be equal to 1
    trial_count++;
    if (trial_count % 22 == 0) {
      //if the trial count is divisible by 22, then run the random trial
      return true;
    } else {
      // otherwise skip the random trial
      return false;
    }
  }
}
</pre>
